libs/api/libbpf_map.cpp

```diff
@@ -408,3 +425,73 @@ libbpf_bpf_map_type_str(enum bpf_map_type t)
 
     return _ebpf_map_display_names[t];
 }
+
+typedef struct perf_buffer
+{
+    std::vector<ebpf_map_subscription_t*> subscriptions;
+} perf_buffer_t;
+
+struct perf_buffer*
+perf_buffer__new(
+    int map_fd,
+    size_t /* page_cnt*/,
+    perf_buffer_sample_fn sample_cb,
+    perf_buffer_lost_fn lost_cb,
+    void* ctx,
+    const struct perf_buffer_opts* /* opts*/)
```

Comment:
If you ever want to support this in the future, then you cannot ignore it now you have to check for must-be-null.  Maybe the same for page_cnt too.  Compare the implementation of perf_buffer__new in external\bpftool\libbpf\src\libbpf.c